TypeCobol/Compiler/CompilationDocument.cs

```diff
@@ -513,6 +506,28 @@ private void ApplyTextChange(TextChange textChange, IList<DocumentChange<ICobolT
             }
         }
 
+        private void AdjustLines(int startIndex, int offset)
+        {
+            bool encounteredCodeElement = false; //Will allow to update line index without erasing all diagnostics after the first encountered line with CodeElements
+            using (var enumerator = compilationDocumentLines.GetEnumerator(startIndex, -1, false))
+            {
+                //Loop on every line that appears after target line
+                CodeElementsLine lineToUpdate;
+                while (enumerator.MoveNext() && (lineToUpdate = enumerator.Current) != null)
+                {
+                    //Remove generated diagnostics for the target line and below.
+                    if (!encounteredCodeElement)
+                        lineToUpdate.ResetDiagnostics(); //Reset diags when on the same zone
+
+                    if (offset != 0)
+                        lineToUpdate.Shift(offset);
+
+                    if (lineToUpdate.CodeElements != null)
+                        encounteredCodeElement = true;
+                }
```

Comment:
No need to iterate over all lines if there is nothing more to do
```suggestion
                    if (offset == 0 && encounteredCodeElement)
                        break; //Nothing more to do
                }
```